progress.js

```diff
@@ -0,0 +1,14 @@
+class Progress {
+  constructor(models) {
+    this.models = models;
+  }
+  async compute(spec) {
+    const { StatusCounter, Status } = this.models;
+    const [{ count }, { total }] = await Promise.all([StatusCounter, Status].map(m => m.findOne(spec)));
```

Comment:
Why do this serially?

Reply:
@indexzero Promise.all is in parallel, Im not sure what you mean here.